lib/Pool.js

```diff
@@ -255,6 +259,13 @@ Pool.prototype.escapeId = function escapeId(value) {
   return mysql.escapeId(value, false);
 };
 
+Pool.prototype.format = function(sql, values) {
+  if (typeof this.config.connectionConfig.queryFormat == "function") {
+    return this.config.connectionConfig.queryFormat.call(this, sql, values, this.config.connectionConfig.timezone);
```

Comment:
I did find a solution to keep `createQuery`  as a static method of `Connection` and provide the right context for instance config, which keeps things backward compatible. I'll commit the changes later from another machine.